### PR TITLE
[INFRA-351] Jenkinsfile for code check jobs

### PR DESCRIPTION
**Pull request by relyafi**
Simple copy of the existing TC "Code checks" job.

Tested the failure case by injecting an unused variable into the code.

### CODE REVIEW

File: .ci/dev/pr-code-checks/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    stages {
+        stage('Detekt check') {
+            steps {
+                script {
+                    pullRequest.createStatus(
+                            status: 'pending',
+                            context: "${PR_CONTEXT_STRING}",
+                            description: "Running code checks",
+                            targetUrl: "${env.BUILD_URL}")
```
Comment (wzur-r3) [inline .ci/dev/pr-code-checks/Jenkinsfile:25]: How a build for this Jenkinsfile is started? There is a better way to set custom Github notification context, directly in the configuration of the Github organisation folder

Comment (relyafi) [reply .ci/dev/pr-code-checks/Jenkinsfile:25]: It triggers automatically through the trigger mechanism, in the same way as the mandatory tests.

I wasn't aware we could do that - I don't have admin access to Github so I don't think I can do this myself. FWIW the on-demand tests use this approach too, so we should probably standardise on an approach (and naming convention) for all checks.

Comment (wzur-r3) [reply .ci/dev/pr-code-checks/Jenkinsfile:25]: > I wasn't aware we could do that - I don't have admin access to Github

There is a Jenkins plugin, which has been already installed and it allows to change the Github notification context from its default value to anything you like. It is available as a trait in the configuration of the source in the Github organisation plugin in Jenkins.

**Review by wzur-r3 — APPROVED** [review]
LGTM

**Pull request merged**